Source/Core/Core/IOS/WFS/WFSI.cpp

```diff
@@ -565,13 +565,13 @@ u32 WFSIDevice::GetTmd(u16 group_id, u32 title_id, u64 subtitle_id, u32 address,
     WARN_LOG_FMT(IOS_WFS, "GetTmd: no such file or directory: {}", path);
     return WFS_ENOENT;
   }
+  *size = static_cast<u32>(fp.GetSize());
   if (address)
   {
     auto& system = GetSystem();
     auto& memory = system.GetMemory();
-    fp.ReadBytes(memory.GetPointer(address), fp.GetSize());
+    fp.ReadBytes(memory.GetPointerForRange(address, *size), *size);
```

Comment:
This looks rather spooky, is that really what the console does here, just read the file into memory regardless of its filesize??

Reply:
TMDs are supposed to always be a fixed size, so I wouldn't be surprised if the real IOS had some kind of size check here... But I don't know if it does or not.